From 77a2c2478712d3aa3a3381313ea6ab314b0ad8bc Mon Sep 17 00:00:00 2001
From: Oleg Karfich <oleg.karfich@wago.com>
Date: Fri, 29 Jun 2018 14:26:12 +0200
Subject: [PATCH 028/144] board: wago-pfc-am335x: set ethaddr also on eth1

Ethernet device iteration during probing requires strict dependencies between
ethernet alias and rmmi interface. Due to this restriction we also need to set
the ethaddr on eth1 device wich belongs in kernel to eth0 and can be parsed in
kernel from oftree.

Signed-off-by: Oleg Karfich <oleg.karfich@wago.com>
---
 arch/arm/boards/wago-pfc-am335x/board.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boards/wago-pfc-am335x/board.c b/arch/arm/boards/wago-pfc-am335x/board.c
index ddcd19d..5ccb3c1 100644
--- a/arch/arm/boards/wago-pfc-am335x/board.c
+++ b/arch/arm/boards/wago-pfc-am335x/board.c
@@ -37,7 +37,8 @@
 #include <globalvar.h>
 #include <xsection.h>
 
-#define ETH_DEVICE 0
+#define ETH_DEVICE_0 0
+#define ETH_DEVICE_1 1
 
 static int pfc_coredevice_init(void)
 {
@@ -78,7 +79,8 @@ static int pfc_set_ethaddr(void)
 	if (ret < 0)
 		return ret;
 
-	eth_register_ethaddr(ETH_DEVICE, mac);
+	eth_register_ethaddr(ETH_DEVICE_0, mac);
+	eth_register_ethaddr(ETH_DEVICE_1, mac);
 
 	return 0;
 }
-- 
2.7.4

